modules/solutions/twitter-analytics/siddhi-extensions/siddhi-execution-tokenizer/component/src/main/java/org/wso2/extension/siddhi/execution/tokenizer/TweetTextTokenizer.java

```diff
@@ -0,0 +1,183 @@
+                + "|" + decorations);
+        while (streamEventChunk.hasNext()) {
+            StreamEvent streamEvent = streamEventChunk.next();
+            String event = (String) attributeExpressionExecutors[0].execute(streamEvent);
+            event = removeEmojis(event);
+            event = event.replaceAll(urlPattern, "").replaceAll("@(.*)", "").
+                    replaceAll("#(.*)", "").replaceAll("[0-9]+", "")
+                    .replaceAll("‼", "").replaceAll("…", "");
+            String[] words = pattern.split(event);
+            for (String word : words) {
+                if (!word.equals("") && isMeaningful(word) && word.length() > 2) {
+                    Object[] data = {word};
+                    complexEventPopulater.populateComplexEvent(streamEvent, data);
+                    nextProcessor.process(streamEventChunk);
+                }
+            }
+        }
+    }
+
+    /**
+     * The initialization method for {@link StreamProcessor}, which will be called before other methods and validate
+     * the all configuration and getting the initial values.
+     *
+     * @param attributeExpressionExecutors are the executors of each attributes in the Function
+     * @param configReader                 this hold the {@link StreamProcessor} extensions configuration reader.
+     * @param siddhiAppContext             Siddhi app runtime context
+     */
+    @Override
+    protected List<Attribute> init(AbstractDefinition inputDefinition,
+                                   ExpressionExecutor[] attributeExpressionExecutors, ConfigReader configReader,
+                                   SiddhiAppContext siddhiAppContext) {
+        if (attributeExpressionExecutors.length == 1) {
+            if (attributeExpressionExecutors[0].getReturnType() != Attribute.Type.STRING) {
+                throw new SiddhiAppCreationException("Text should be of type string. But found "
+                        + attributeExpressionExecutors[0].getReturnType());
+            }
+        } else {
+            throw new IllegalArgumentException(
+                    "Invalid no of arguments passed to text:tokenize() function, "
+                            + "required 1, but found " + attributeExpressionExecutors.length);
+        }
+
+        List<Attribute> attributes = new ArrayList<>();
+        attributes.add(new Attribute("token", Attribute.Type.STRING));
+        return attributes;
+    }
+
+    @Override
+    public void start() {
+        //Do Nothing
+    }
+
+    @Override
+    public void stop() {
+        //Do Nothing
+    }
+
+    @Override
+    public Map<String, Object> currentState() {
+        return null;
+    }
+
+    @Override
+    public void restoreState(Map<String, Object> state) {
+        //Do Nothing
+    }
+
+    private boolean isMeaningful(String word) {
+        String line;
+        InputStream inputStream = TweetTextTokenizer.class.getResourceAsStream("/words.csv");
```

Comment:
Why are you loading this words.csv per each event? Why is it not possible to load it at the extension init?